job/server/src/main/java/alluxio/job/transform/format/csv/CsvReader.java

```diff
@@ -46,27 +46,27 @@
   private final CsvSchema mSchema;
 
   /**
-   * @param fs the hdfs compatible client
    * @param inputPath the input path
    * @throws IOException when failed to open the input
    */
-  private CsvReader(FileSystem fs, Path inputPath) throws IOException {
+  private CsvReader(Path inputPath) throws IOException {
     mCloser = Closer.create();
-    mFs = fs;
+    Configuration conf = ReadWriterUtils.readNoCacheConf();
+    mFs = inputPath.getFileSystem(conf);
```

Comment:
You can do `mFs = mCloser.register(inputPath.getFileSystem(conf))` (applies to `mReader` too)

Reply:
@chcn you don't like this?